### DISCUSSION TITLE
How do I update an implicit many-to-many relationship via a one-to-many?

### DISCUSSION THREAD

**Discussion by sjsanc** [prisma/prisma#21039] [Q&A]
### Question

I have three models

```prisma
model Person {
    uuid        String    @id @default(uuid())

    firstName   String
    middleNames String?
    lastName    String
    bornOn      DateTime
    diedOn      DateTime?
    nationality String
    hasCoached  Boolean?

    provinceId Int?
    province   Province? @relation(fields: [provinceId], references: [id], onDelete: SetNull)

    sports  Sport[]
    results Result[]
    events  Event[]   @relation("EventToPerson")

    createdAt DateTime  @default(now())
    updatedAt DateTime? @updatedAt
}

model Event {
    uuid String @id @default(uuid())

    name      String
    startDate DateTime
    endDate   DateTime?

    results Result[]
    tags   Tag[]
    participants Person[] @relation("EventToPerson")

    provinceId Int?
    province   Province? @relation(fields: [provinceId], references: [id], onDelete: SetNull)

    poolUuid String?
    pool     Pool?   @relation(fields: [poolUuid], references: [uuid], onDelete: SetNull)

    createdAt DateTime  @default(now())
    updatedAt DateTime? @updatedAt
}

model Result {
    uuid String @id @default(uuid())

    time     Int
    isRecord Boolean?
    medal    String?

    sportId Int?
    sport   Sport? @relation(fields: [sportId], references: [id])

    personUuid String?
    person   Person? @relation(fields: [personUuid], references: [uuid], onDelete: SetNull)

    eventUuid String?
    event   Event? @relation(fields: [eventUuid], references: [uuid], onDelete: SetNull)

    createdAt DateTime  @default(now())
    updatedAt DateTime? @updatedAt
}
```

This schema produces four tables, with an implicit `_EventToPerson` join table. 

After making my `Person` & `Event`, I want to create a `Result`. (I'm passing the FK directly as per https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#connect - using `connect` doesn't change anything).

```javascript
  const result = await prisma.result.create({
    data: {
      time: body.time,
      isRecord: body.isRecord,
      medal: body.medal,
      eventUuid: body.eventUuid,
      personUuid: body.personUuid,
    },
  });
```

The `create` succeeds in creating a `Result` but I was expecting a record to also be created in the join table. Have I misunderstood how m-n relations work? 

Ultimately I'd like to be able to query a count of `Person` that has a `Result` connected to a particular `Event`, like so: 

```js
result = await prisma.event.findMany({
    include: {
      _count: {
        select: { participants: true },
      },
    },
  });
  ```

### Information about Prisma Schema, Client Queries and Environment (optional)

 "prisma": "^5.2.0",

**Comment by ludralph** [comment]
Hi @sjsanc 👋 

Thank you for raising this question.

> The create succeeds in creating a Result but I was expecting a record to also be created in the join table. Have I misunderstood how m-n relations work?

For Implicit many to many relations, the relation table is maintained for you under the hood by Prisma so you need not worry about it.

> Ultimately I'd like to be able to query a count of Person that has a Result connected to a particular Event, like so:

To query a count of `Person` that has a `Result` connected to a particular Event, you can use the _count parameter with a nested select in Prisma. For example:
```js
const eventUuid = 'your-event-uuid'; // replace with your event UUID
const personCount = await prisma.person.findMany({
  where: {
    results: {
      some: {
        event: {
          uuid: eventUuid 
        }
      }
    }
  },
  select: {
    uuid: true,
    firstName: true,
    _count: {
      select: { results: true },
    },
  },
});
```
This query will return a list of Person objects that have a Result connected to the specified Event, along with a count of their `Result` objects.
The output will look like:
```js
[
  {
    uuid: '9cba986b-22f9-41f6-90d0-e26f07ae4c64',
    firstName: 'Alice',
    _count: { results: 2 }
  },
  {
    uuid: '6ec6a602-0d83-49ee-bd60-4e44890bd755',
    firstName: 'John',
    _count: { results: 1 }
  }
]
```
So, for example, the first object in the array:
```js
{
  uuid: '9cba986b-22f9-41f6-90d0-e26f07ae4c64',
  firstName: 'Alice',
  _count: { results: 2 }
}
```
This tells us that the person with the `uuid` of `9cba986b-22f9-41f6-90d0-e26f07ae4c64` and the first name `Alice` has 2 `Result` records associated with her.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by nikolasburk** [reply]
@sjsanc It seems like you have _two_ many-to-many relations in your schema:

- the _implicit_ m-n relation you've pointed out yourself, which results in the `_EventToPerson` relation table
- an _explicit_ m-n relation where `Result` actually _is_ the relation table which has a couple of additional fields as well to store "metadata" about the relationship

My hunch would be that you may be able to drop the implicit m-n relation to simplify your DB schema and the queries you're sending, unless there's a dedicated reason you need it?!

**Reply by sjsanc** [reply]
Hey folks. I appreciate your help!

@nikolasburk - I don't think I can use the Results table, as there could be many results per person per event, so the PK wouldn't be unique. Though perhaps I could add a third ID to the compound PK?
